css-scroll-snap-2/Overview.bs

```diff
@@ -275,119 +275,209 @@ Snap Events {#snap-events}
+		<a spec="css-scroll-snap-1" lt="scroll snap area">snap areas</a>.
+	</li>
+	<li> if there is a change to a snap container's style such that it goes from
+		having a non-'none' value for [[css-scroll-snap-1#scroll-snap-type|scroll-snap-type]]
+		to having a 'none' value or vice versa.
+	</li>
+	<li> if, after a [[css-scroll-snap-1#re-snap|layout change]], the element to
+		which a snap container is	snapped to changes, regardless of whether there is
+		a change in scroll position	after the layout change.
+	</li>
+	</ol>
+
+	Although, snapping occurs when a snap container is first laid out, {{snapchanged}}
+	should not be triggered by this initial layout.
+
+	Scrolling operations always lead to {{scrollend}} events being fired. If a
+	scrolling operation also results in a {{snapchanged}} event being fired, the
+	{{snapchanged}} event should be fired before the {{scrollend}} event.
+
+	Each
+	<a spec=css-scroll-snap lt="scroll snap container">snap container</a> is
+	associated with at most one <dfn export>snapchangedTargetBlock</dfn> and at most one
+	<dfn export>snapchangedTargetInline</dfn> in the block and inline axes respectively.
+
+	When asked to <dfn export for=Document>update snapchanged targets</dfn>
+	for a <a spec=css-scroll-snap lt="scroll snap container">snap container</a>,
+	|snapcontainer|, run these steps:
+
+	1. Let <var>doc</var> be |snapcontainer|'s associated {{Document}}.
+	1. Let <var>blockTarget</var> be the <a>snapchangedTargetBlock</a> associated
+		with |snapcontainer|.
+	1. Let <var>inlineTarget</var> be the <a>snapchangedTargetInline</a> associated
+		with |snapcontainer|.
+	1. Let <var>blockSnapchangingTarget</var> be the <a>snapchangingTargetBlock</a>
+		associated with |snapcontainer|.
+	1. Let <var>inlineSnapchangingTarget</var> be the
+		<a>snapchangingTargetInline</a> associated with |snapcontainer|.
+	1. If <var>blockTarget</var> is not the same element as
+		<var>blockSnapchangingTarget</var>:
+		1. Set the <a>snapchangedTargetBlock</a> associated with |snapcontainer| to
+			<var>blockSnapchangingTarget</var>.
+		1. If |snapcontainer| is not already in <var>doc</var>'s
+			<a spec=cssom-view lt="pending snapchanged event targets">pending snapchanged event targets</a>.
+			1. Append |snapcontainer| to <var>doc</var>'s
+				<a spec=cssom-view lt="pending snapchanged event targets">pending snapchanged event targets</a>.
+	1. If <var>inlineTarget</var> is not the same element as
+			<var>inlineSnapchangingTarget</var>:
+		1. Set the <a>snapchangedTargetInline</a> associated with |snapcontainer| to
+			<var>inlineSnapchangingTarget</var>.
+		1. If |snapcontainer| is not already in <var>doc</var>'s
+			<a spec=cssom-view lt="pending snapchanged event targets">pending snapchanged event targets</a>.
+			1. Append |snapcontainer| to <var>doc</var>'s
+				<a spec=cssom-view lt="pending snapchanged event targets">pending snapchanged event targets</a>.
```

Comment:
It took me some careful reading to work out why we can use the blockSnapChangingTarget here. Can you add a note below this explaining that the snapchanging inline and block targets are updated before snapchanged and as such represent the actual new snap target?